Git-commit: d1f6bebac64d1c300d2ecf103db092f0c1d6c1d8
From: Wei Li <liwei391@huawei.com>
Date: Tue, 23 Feb 2021 20:16:36 +0800
Subject: [PATCH] arm64: add new config CONFIG_PMU_WATCHDOG
Patch-mainline: Queued in openEuler repo, version 5.10.0-4.2.0
References: openEuler-21.03
Git-repo: https://gitee.com/openeuler/kernel.git

hulk inclusion
category: feature
bugzilla: 49592
CVE: NA

-------------------------------------------------

Add new config CONFIG_PMU_WATCHDOG for watchdog implementation method
configuration.

Signed-off-by: Wei Li <liwei391@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Kai Liu <kai.liu@suse.com>
---
 arch/arm64/Kconfig |  1 -
 lib/Kconfig.debug  | 13 +++++++++++++
 2 files changed, 13 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/Kconfig b/arch/arm64/Kconfig
index 446cf0206c64..ce0e131b3c85 100644
--- a/arch/arm64/Kconfig
+++ b/arch/arm64/Kconfig
@@ -172,7 +172,6 @@ config ARM64
 	select HAVE_NMI
 	select HAVE_PATA_PLATFORM
 	select HAVE_PERF_EVENTS
-	select HAVE_HARDLOCKUP_DETECTOR_PERF	if PERF_EVENTS && HAVE_PERF_EVENTS_NMI
 	select HAVE_PERF_REGS
 	select HAVE_PERF_USER_STACK_DUMP
 	select HAVE_REGS_AND_STACK_ACCESS_API
diff --git a/lib/Kconfig.debug b/lib/Kconfig.debug
index 7dd44d922ea9..a7dd39c792e0 100644
--- a/lib/Kconfig.debug
+++ b/lib/Kconfig.debug
@@ -964,12 +964,25 @@ config HARDLOCKUP_DETECTOR_PERF
 	bool
 	select SOFTLOCKUP_DETECTOR
 
+choice
+	prompt "aarch64 NMI watchdog method"
+	depends on ARM64
+	help
+	  Watchdog implementation method configuration.
+
 config SDEI_WATCHDOG
 	bool "SDEI NMI Watchdog support"
 	depends on ARM_SDE_INTERFACE && !HARDLOCKUP_CHECK_TIMESTAMP
 	select HAVE_HARDLOCKUP_DETECTOR_ARCH
 	select HARDLOCKUP_DETECTOR
 
+config PMU_WATCHDOG
+	bool "PMU NMI Watchdog support"
+	depends on PERF_EVENTS && HAVE_PERF_EVENTS_NMI
+	select HAVE_HARDLOCKUP_DETECTOR_PERF
+
+endchoice
+
 #
 # Enables a timestamp based low pass filter to compensate for perf based
 # hard lockup detection which runs too fast due to turbo modes.
-- 
2.31.1

